contracts/mocks/finance/VestingWalletConfidentialFactoryMock.sol

```diff
@@ -12,14 +12,32 @@ abstract contract VestingWalletConfidentialFactoryMock is VestingWalletConfident
         return address(new VestingWalletCliffExecutorConfidential());
     }
 
+    function _validateVestingWalletInitialization(bytes memory initialization) internal virtual override {
+        // solhint-disable no-unused-vars
+        (
+            address beneficiary,
+            uint48 startTimestamp,
+            uint48 durationSeconds,
+            uint48 cliffSeconds,
+            address executor
+        ) = abi.decode(initialization, (address, uint48, uint48, uint48, address));
+
+        require(cliffSeconds <= durationSeconds);
+        require(beneficiary != address(0));
```

Comment:
Should we:
1. provide this as ther default behavior in the factory
2. and same with `return address(new VestingWalletCliffExecutorConfidential());`
3. And add that piece of code to documentation:
```
    function _deployVestingWalletImplementation() internal virtual override returns (address) {
        return address(new MyCustomVestingWalletConfidential());
    }

    function _validateVestingWalletInitialization(bytes memory initialization) internal virtual override {
        (
            address arg1,
            uint48 arg2,
        ) = abi.decode(initialization, (address, uint48));
        require(arg1 != address(0));
    }
```

Reply:
You are right about config yes, hence keeping both is best yes 👍